sklearn/datasets/tests/test_kddcup99.py

```diff
@@ -44,3 +45,37 @@ def test_shuffle(fetch_kddcup99_fxt):
     dataset = fetch_kddcup99_fxt(random_state=0, subset='SA', shuffle=True,
                                  percent10=True)
     assert(any(dataset.target[-100:] == b'normal.'))
+
+
+def test_asframe(fetch_kddcup99_fxt):
```

Comment:
You can parametrize the test here using `pytest.mark.parametrize` and passing the parameter `subset` and the expected shape of the data.

A better for this function would be `test_fetch_kddcup99_check_as_frame_shape`

Reply:
since the amount of parameters and checks are relatively large, it's just gonna make the decorator very verbose, pretty much just as verbose as the current function itself. Don't really see the benefit of using pytest.mark.parametrize here?